Workflow API docs

hhunter-ms opened the pull request:
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Chalking out workflow API docs

## Issue reference

PR will close: #2798 and #2933

hhunter-ms committed:
start workflow api skeleton

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
updates to overview

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
updates to overview

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
update overview, add to how-to

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
add scenarios

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
update to overview

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
diagrams

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
update to overview

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
move things around

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
update overview

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
add new file

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
move things around per nyemade

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

RyanLettieri commented on daprdocs/content/en/operations/components/workflow-engine/workflow-engine.md:
```diff
@@ -0,0 +1,82 @@
+The workflow building block consists of:
+
+- A pluggable component model for integrating various workflow engines
+- A set of APIs for managing workflows (start, schedule, pause, resume, cancel)
```
Getting information on a workflow is another capability of the current API. If it feels like a "management" capability, we can put it in this list

hhunter-ms committed:
updates

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
rearrange how to

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
remove links for now

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

nyemade-uversky commented on daprdocs/content/en/concepts/building-blocks-concept.md:
```diff
@@ -20,12 +20,13 @@ The following are the building blocks provided by Dapr:
+| [**Secrets**]({{< ref "secrets-overview.md" >}}) | `/v1.0/secrets` | Dapr provides a secrets building block API and integrates with secret stores such as public cloud stores, local stores and Kubernetes to store the secrets. Services can call the secrets API to retrieve secrets, for example to get a connection string to a database.
+| [**Configuration**]({{< ref "configuration-api-overview.md" >}}) | `/v1.0-alpha1/configuration` | The Configuration API enables you to retrieve and subscribe to application configuration items for supported configuration stores. This enables an application to retrieve specific configuration information, for example, at start up or when configuration changes are made in the store. 
+| [**Distributed lock**]({{< ref "distributed-lock-api-overview.md" >}}) | `/v1.0-alpha1/lock` | The distributed lock API enables you to take a lock on a resource so that multiple instances of an application can access the resource without conflicts and provide consistency guarantees.  
+| [**Workflows**]({{< ref "workflow-overview.md" >}}) | `/v1.0-alpha1/workflow` | The workflow API enables you to define a processes or data flows that span multiple microservices via an embedded workflow engine. With this built-in workflow engine, you can easily integrate with existing Dapr building blocks while maintaining portability.
```
```suggestion
| [**Workflows**]({{< ref "workflow-overview.md" >}}) | `/v1.0-alpha1/workflow` | The Workflow API enables you to define processes or data flows that span multiple microservices via the dapr-embedded workflow engine or supported external workflow components. With the Workflow API, you can easily integrate with existing Dapr building blocks while maintaining portability.
```

nyemade-uversky commented on daprdocs/content/en/concepts/components-concept.md:
```diff
@@ -101,9 +101,16 @@ Lock components are used as a distributed lock to provide mutually exclusive acc
 
+### Workflows
+
+A [workflow]({{< ref workflow-overview.md >}}) is custom application logic that defines business process or data flow in a reliable way across mulitple microservices. The workflow API is exposed by a [lightweight, embedded workflow engine]({{< ref "operations/components/workflow-engine/workflow-engine.md" >}}) in the Dapr sidecar, allowing you to easily integrate with existing Dapr building block ...
```
> The workflow API is exposed by a [lightweight, embedded workflow engine]({{< ref "operations/components/workflow-engine/workflow-engine.md" >}}) in the Dapr sidecar, allowing you to easily integrate with existing Dapr building blocks

@RyanLettieri Could you add more context here? Is the workflow API exposed through the workflow engine - will users need the workflow engine in place to use the workflow API (even if with Temporal)?

msfussell replied:
This section is about workflow components, of which there is one and that is Temporal. 

```suggestion
A [workflow]({{< ref workflow-overview.md >}}) is custom application logic that defines a reliable business process or data flow. Workflow components are workflow runtimes (or engines) that run the business logic written for that workflow and store their state into a state store. 
```

nyemade-uversky commented on daprdocs/content/en/concepts/components-concept.md:
```diff
@@ -101,9 +101,16 @@ Lock components are used as a distributed lock to provide mutually exclusive acc
+
+A [workflow]({{< ref workflow-overview.md >}}) is custom application logic that defines business process or data flow in a reliable way across mulitple microservices. The workflow API is exposed by a [lightweight, embedded workflow engine]({{< ref "operations/components/workflow-engine/workflow-engine.md" >}}) in the Dapr sidecar, allowing you to easily integrate with existing Dapr building block ...
+
+- [List of supported workflows]({{< ref supported-workflows >}})
```
```suggestion
- [List of supported workflow components]({{< ref supported-workflows >}})
```

nyemade-uversky commented on daprdocs/content/en/developing-applications/building-blocks/workflow/_index.md:
```diff
@@ -0,0 +1,7 @@
+title: "Workflow engines"
+linkTitle: "Workflow"
+weight: 100
+description: Expose workflows as APIs with HTTP and Dapr SDK support
```
```suggestion
title: "Workflows"
linkTitle: "Workflows"
weight: 100
description: Create and run workflows for custom application logic or data flows across microservices. 
```

hhunter-ms commented:
#2933

hhunter-ms committed:
some edits

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
Merge branch 'v1.10' into issue_2798b

Signed-off-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>

hhunter-ms committed:
initial edits from nyemade

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

nyemade-uversky commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,111 @@
+---
+type: docs
+title: Workflow overview
+linkTitle: Overview
+weight: 1000
+description: "Overview of the workflow building block"
+---
+
+Workflows are quickly becoming a core element to cloud-native applications. A workflow is custom application logic that:
+
+- Defines a business process or data flow that spans multiple microservices
+- Drives the execution of these business processes to completion in a reliable way
+- Consists of a set of tasks and/or state transitions, which can be either statically defined or dynamic
+- May be authored using declarative markup or general purpose programming languages
+
```
Reworked this section a bit.

nyemade-uversky committed:
Changed workflow overview

nyemade-uversky committed:
Update overview

nyemade-uversky committed:
Re-worked features list

hhunter-ms committed:
Merge branch 'v1.10' of https://github.com/dapr/docs into issue_2798b

hhunter-ms committed:
fix link

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

cgillum committed:
Add outline for workflow engine operational doc

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

cgillum committed:
Initial workflow engine operational content

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

nyemade-uversky committed:
Updated how to

nyemade-uversky committed:
Remove logic apps

RyanLettieri commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,94 @@
+### Combine Dapr APIs
+- Workflows allows users to combine Dapr APIs together in scheduled tasks
+- Workflows compatible with pubsub, state store, bindings 
+- Can send/respond to external signals, including pub/sub events and bindings (input or output
```
Nit: Close the parentheses at the end of this. Also, start the sentence with "Workflows" to be consistent with the two points above.

RyanLettieri commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,94 @@
+
+### Combine Dapr APIs
+- Workflows allows users to combine Dapr APIs together in scheduled tasks
+- Workflows compatible with pubsub, state store, bindings 
```
"Workflows are compatible"?

nyemade-uversky committed:
Remove logic apps

nyemade-uversky committed:
Delete dtfx-go.md

nyemade-uversky committed:
Update workflow api reference page

nyemade-uversky committed:
update metadata attribute table

nyemade-uversky committed:
update response body for start method

nyemade-uversky committed:
update post and get request

nyemade-uversky committed:
update how to section

cgillum committed:
More Dapr workflow operational updates

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

cgillum committed:
Merge branch 'issue_2798b' of https://github.com/hhunter-ms/docs into issue_2798b

hhunter-ms committed:
update building blocks image

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
update images

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
missed an image

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
remove incomplete link

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
match colors in images

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
add workflows to the faq

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
remove logic apps and dtfx-go

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
add as preview feature with quick grammar pass

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
add authoring workflow doc to components section

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
add features and workflow patterns

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
add features header back to overview doc

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

RyanLettieri commented on daprdocs/content/en/concepts/faq.md:
```diff
@@ -27,6 +27,16 @@ Creating a new actor follows a local call like `http://localhost:3500/v1.0/actor
+
+### Differences between Dapr Workflows, authoring APIs, and the management API
+
+TODO
```
There are 2 sides of the workflow SDK: The authoring portion, and the management portion. The management side of the SDK allows you to start, stop, and get information on a workflow. This can be achieved through simple HTTP or gRPC commands.

RyanLettieri commented on daprdocs/content/en/developing-applications/building-blocks/workflow/howto-workflow.md:
````diff
@@ -0,0 +1,95 @@
+```bash
+command to register Temporal workflow for all steps involved
+```
+
````
First, a yaml file for temporal must be created. You can find an example here: <TODO: Link yaml file>.
Next, the temporal server must be started. This can be done from a docker image in combination with the "docker compose up" command.
Once the yaml file has been created, and the server has been started, all you need to do is add one additional line into the normal dapr command and dapr will discover the component: <“--components-path <path to temporal component yaml>”>

hhunter-ms replied:
saving this blurb for when temporal is added and resolving this comment

RyanLettieri committed:
Uploading workflow api diagram

hhunter-ms committed:
update patterns text

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
update next steps

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

RyanLettieri commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,49 @@
+
+*To Do:*
+- *Add diagram of how workflow API works?*
+- *Add explaination of how the API works*
```
<img src="/dapr/docs/daprdocs/static/images/workflows-diagram.png" width=1000>
Create an application and run it with the dapr sidecar. After running the application, HTTP/gRPC calls can be made to dapr to run specific tasks/workflows that reside in the application.

Each individual workflow can be started or terminated through a POST request. You can also get information on the workflow (even if it has been terminated or finished naturally!) through a GET request. This GET request will send back information such as the instance ID of the workflow as well as the time that the run started, and even the current running status whether that be “Running”, “Terminated” or “Completed”.

RyanLettieri commented on daprdocs/content/en/developing-applications/develop-components/develop-workflows.md:
```diff
@@ -0,0 +1,138 @@
+
+## Currently supported SDK languages
+
+TODO
```
Currently, you can use the following SDK languages to manage a workflow:

- [.NET SDK]

msfussell commented on daprdocs/content/en/concepts/building-blocks-concept.md:
```diff
@@ -20,12 +20,13 @@ The following are the building blocks provided by Dapr:
+| [**Secrets**]({{< ref "secrets-overview.md" >}}) | `/v1.0/secrets` | Dapr provides a secrets building block API and integrates with secret stores such as public cloud stores, local stores and Kubernetes to store the secrets. Services can call the secrets API to retrieve secrets, for example to get a connection string to a database.
+| [**Configuration**]({{< ref "configuration-api-overview.md" >}}) | `/v1.0-alpha1/configuration` | The Configuration API enables you to retrieve and subscribe to application configuration items for supported configuration stores. This enables an application to retrieve specific configuration information, for example, at start up or when configuration changes are made in the store. 
+| [**Distributed lock**]({{< ref "distributed-lock-api-overview.md" >}}) | `/v1.0-alpha1/lock` | The distributed lock API enables you to take a lock on a resource so that multiple instances of an application can access the resource without conflicts and provide consistency guarantees.  
+| [**Workflows**]({{< ref "workflow-overview.md" >}}) | `/v1.0-alpha1/workflow` | The Workflow API enables you to define processes or data flows that span multiple microservices via the Dapr-embedded workflow engine or supported external workflow components. With the Workflow API, you can easily integrate with existing Dapr building blocks while maintaining portability.
```
```suggestion
| [**Workflows**]({{< ref "workflow-overview.md" >}}) | `/v1.0-alpha1/workflow` | The Workflow API enables you to define long running, persistent processes or data flows that span multiple microservices using Dapr workflows or workflow components. The Workflow API can be combined  with other Dapr API building blocks, for example a workflow can call another service with service invocation, or retrieve secrets, providing flexibility and portability. 
```

nyemade-uversky committed:
Update workflkow concepts

nyemade-uversky committed:
update concepts links

nyemade-uversky committed:
update how to

hhunter-ms committed:
Merge branch 'v1.10' into issue_2798b

Signed-off-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>

hhunter-ms committed:
address first Mark comment

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
Merge branch 'issue_2798b' of https://github.com/hhunter-ms/docs into issue_2798b

hhunter-ms committed:
small edits

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
add spaces to ref shortcode

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

msfussell commented on daprdocs/content/en/concepts/components-concept.md:
```diff
@@ -101,9 +101,16 @@ Lock components are used as a distributed lock to provide mutually exclusive acc
+A [workflow]({{< ref workflow-overview.md >}}) is custom application logic that defines business process or data flow in a reliable way across mulitple microservices. The workflow API is exposed by a [lightweight, embedded workflow engine]({{< ref "operations/components/workflow-engine/workflow-engine.md" >}}) in the Dapr sidecar, allowing you to easily integrate with existing Dapr building block ...
+
+- [List of supported workflow components]({{< ref supported-workflows >}})
+- Workflow implementations
```
```suggestion
- [Workflow implementations](- [Binding implementations](https://github.com/dapr/components-contrib/tree/master/workflows)
```

msfussell commented on daprdocs/content/en/concepts/faq.md:
```diff
@@ -27,6 +27,16 @@ Creating a new actor follows a local call like `http://localhost:3500/v1.0/actor
 
 The Dapr runtime SDKs have language-specific actor frameworks. For example, the .NET SDK has C# actors. The goal is for all the Dapr language SDKs to have an actor framework. Currently .NET, Java, Go and Python SDK have actor frameworks.
 
+## Workflows
```
Both of these FAQ can be removed. They are not needed. 
```suggestion

```

msfussell commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,49 @@
+description: "Overview of the workflow building block"
+---
+
+With Dapr workflows, you can automate and orchestrate tasks within your application. The workflow feature provides a developer-friendly programming model for authoring workflows as code in a way that abstracts away the complexities of messaging, state management, and failure handling.
```
```suggestion
With workflows, you can orchestrate tasks or business logic within your application, making them reliable and restartable. Workflows provide a programming model for authoring code that abstracts away the complexities of messaging, state management, and failure handling.
```

msfussell commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,49 @@
+
+With Dapr workflows, you can automate and orchestrate tasks within your application. The workflow feature provides a developer-friendly programming model for authoring workflows as code in a way that abstracts away the complexities of messaging, state management, and failure handling.
+
+The workflow feature is comprised of the Workflow API and the workflow components that allow for authoring workflows naitively in Dapr or through external 3rd party workflow applications.
```
```suggestion
```

msfussell commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,49 @@
+
+The workflow feature is comprised of the Workflow API and the workflow components that allow for authoring workflows naitively in Dapr or through external 3rd party workflow applications.
+
+<img src="/images/concepts-building-blocks.png" width=250>
```
```suggestion

```

msfussell commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,49 @@
+
+<img src="/images/concepts-building-blocks.png" width=250>
+
+## Workflow API
```
```suggestion
## Workflow building block
```

msfussell commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,49 @@
+<img src="/images/concepts-building-blocks.png" width=250>
+
+## Workflow API
+
```
The workflow building block:
- Provides a workflow API for running workflows 
- An built-in workflow runtime to write Dapr workflows (of type `workflow.dapr`)
- Integrates with various workflow runtimes as components, for example Temporal workflows.

msfussell commented on daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-overview.md:
```diff
@@ -0,0 +1,49 @@
+
+## Features
+
+*To Do*
```
```suggestion
Start, Terminate and X any workflow
```

hhunter-ms committed:
small updates per Mark

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

cgillum committed:
Fix image sizes and start workflow development guide

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

cgillum committed:
More workflow authoring content

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

hhunter-ms committed:
update overview w info from Ryan and community call demo

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
Merge branch 'v1.10' into issue_2798b

Signed-off-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>

cgillum committed:
Merge branch 'issue_2798b' of https://github.com/hhunter-ms/docs into issue_2798b

cgillum committed:
More workflow authoring content plus reordering

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

nyemade-uversky committed:
Update howto-workflow.md

cgillum committed:
More reordering/refactoring of develop-workflows.md

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

cgillum committed:
Merge branch 'issue_2798b' of https://github.com/hhunter-ms/docs into issue_2798b

nyemade-uversky committed:
Update workflow-overview.md

nyemade-uversky committed:
Update workflow-overview.md

nyemade-uversky committed:
Update workflow-overview.md

nyemade-uversky committed:
Delete howto-workflow.md

nyemade-uversky committed:
Delete workflow-overview.md

nyemade-uversky committed:
Delete _index.md

nyemade-uversky committed:
Create _index.md

nyemade-uversky committed:
Create workflow-management-overview.md

nyemade-uversky committed:
Update workflow-management-overview.md

hhunter-ms committed:
update how to

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

nyemade-uversky committed:
Update faq.md

hhunter-ms committed:
remove from overview

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
Merge branch 'issue_2798b' of https://github.com/hhunter-ms/docs into issue_2798b

hhunter-ms committed:
updates to how-to, overview, faq

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
revert change so I can run locally

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
fix conflicts

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
updates per sync

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
forgot to add breaking file

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
adding all the files for good measure

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
forgot one lollll

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
fix broken link

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
update diagrams

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

cgillum committed:
Updated workflow-overview.md
- Fixed several terminology issues in the overview
- Rewrote function chaining as "task chaining"
- Added code sample for task chaining
- Added a couple links and diagrams

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

hhunter-ms committed:
add diagrams, quick review of overview

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
pass pt 1

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
update diagrams

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

cgillum committed:
Flesh out fan-out/fan-in description

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

hhunter-ms committed:
workflow architecture readability

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
pass pt 2

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
updates per docs triage meeting with Mark/Nick

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
Merge branch 'v1.10' into issue_2798b

Signed-off-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>

cgillum committed:
Update async HTTP pattern

Signed-off-by: Chris Gillum <cgillum@microsoft.com>

cgillum committed:
Merge branch 'issue_2798b' of https://github.com/hhunter-ms/docs into issue_2798b

hhunter-ms committed:
remove management from language

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

msfussell reviewed (approved):
Let's get this into the docs!!
